snuba/datasets/entities/metrics.py

```diff
@@ -8,20 +8,46 @@
     Nested,
     UInt,
 )
-from snuba.clickhouse.translators.snuba.mappers import ColumnToFunction
+from snuba.clickhouse.translators.snuba.mappers import (
+    ColumnToFunction,
+    SubscriptableMapper,
+)
 from snuba.clickhouse.translators.snuba.mapping import TranslationMappers
 from snuba.datasets.entity import Entity
 from snuba.datasets.plans.single_storage import SingleStorageQueryPlanBuilder
 from snuba.datasets.storages import StorageKey
 from snuba.datasets.storages.factory import get_storage, get_writable_storage
 from snuba.pipeline.simple_pipeline import SimplePipelineBuilder
+from snuba.query.exceptions import InvalidExpressionException
 from snuba.query.expressions import Column as ColumnExpr
+from snuba.query.expressions import Expression, Literal, SubscriptableReference
 from snuba.query.extensions import QueryExtension
+from snuba.query.logical import Query
 from snuba.query.processors import QueryProcessor
 from snuba.query.processors.basic_functions import BasicFunctionsProcessor
 from snuba.query.processors.project_rate_limiter import ProjectRateLimiterProcessor
 from snuba.query.processors.timeseries_processor import TimeSeriesProcessor
 from snuba.query.validation.validators import EntityRequiredColumnValidator
+from snuba.request.request_settings import RequestSettings
+
+
+class TagsTypeTransformer(QueryProcessor):
+    def process_query(self, query: Query, request_settings: RequestSettings) -> None:
+        def transform_expression(exp: Expression) -> Expression:
+            if not isinstance(exp, SubscriptableReference):
+                return exp
+
+            key = exp.key
+            if not isinstance(key.value, str) or not key.value.isdigit():
+                raise InvalidExpressionException(
+                    exp, "Expected a string key in subscriptable"
```

Comment:
Nit: If someone queries with a string key that's not an int, this error message is a little misleading

Reply:
rigght. Will fix.